server/jetstream_cluster_test.go

```diff
@@ -10471,3 +10471,124 @@ func TestJetStreamClusterMirrorDeDupWindow(t *testing.T) {
 	send(100)
 	check(200)
 }
+
+// We had a scenario where a consumer would not recover properly on restart due to
+// the cluster state not being set properly when checking source subjects.
+func TestJetStreamClusterStateOnRestartPreventsConsumerRecovery(t *testing.T) {
+	sc := createJetStreamTaggedSuperCluster(t)
```

Comment:
Since this looks like a super cluster test, and assuming that you are up-to-date with main branch, would you mind moving this test in jetstream_super_cluster_test.go (at the end) and call it TestJetStreamSuperClusterStateOnRestartPreventsConsumerRecovery?

Reply:
Yep.. Will do.